include/podio/Frame.h

```diff
@@ -391,11 +391,13 @@ const CollT& Frame::put(CollT&& coll, const std::string& name) {
 
 template <typename FrameDataT>
 Frame::FrameModel<FrameDataT>::FrameModel(std::unique_ptr<FrameDataT> data) :
-    m_mapMtx(std::make_unique<std::mutex>()),
-    m_data(std::move(data)),
-    m_dataMtx(std::make_unique<std::mutex>()),
-    m_idTable(std::move(m_data->getIDTable())),
-    m_parameters(std::move(m_data->getParameters())) {
+    m_mapMtx(std::make_unique<std::mutex>()), m_dataMtx(std::make_unique<std::mutex>()) {
+  if (!data) {
+    throw std::invalid_argument("Building a Frame failed; if you are reading from a file it may be corrupted");
```

Comment:
```suggestion
    throw std::runtime_error("Building a Frame failed; if you are reading from a file it may be corrupted");
```
Or this, initially I was going to have a message about FrameModel but no one will know what that is so it would be a bit useless.